### PR TITLE
Issue #54: Drop PHP 5.6 support

### LINKED ISSUE #54: Drop PHP 5.6 support

### PR DESCRIPTION
**Pull request by tim-bezhashvyly**
Closes #54.

### CODE REVIEW

File: src/SolrDocumentBuilder.php [hunk 1]
```diff
@@ -31,7 +33,7 @@ public static function fromSearchDocument(SearchDocument $document)
      * @param SearchDocumentFieldCollection $fieldCollection
      * @return array[]
      */
-    private static function getSearchDocumentFields(SearchDocumentFieldCollection $fieldCollection)
+    private static function getSearchDocumentFields(SearchDocumentFieldCollection $fieldCollection) : array
     {
         return array_reduce($fieldCollection->getFields(), function ($carry, SearchDocumentField $field) {
             return array_merge([$field->getKey() => $field->getValues()], $carry);
```
Comment (Vinai) [inline src/SolrDocumentBuilder.php]: Just noticed, merging `$carry` into `[$field->getKey() => $field->getValues()]` would overwrite the new value if `$carry` already contains the same key. Is that behavior ("first one wins") better then the other "last one wins"? Probably it doesn't matter, just wondering what would be least surprising.
This comment applies to several `array_reduce` in this PR.

File: src/SolrResponse.php [hunk 2]
```diff
@@ -111,7 +110,7 @@ private function getNonSelectedFacetFieldsFromSolrFacetQueries(array $selectedFi
     /**
      * @return array[]
      */
-    private function getFacetFields()
+    private function getFacetFields() : array
     {
```
Comment (Vinai) [inline src/SolrResponse.php]: Null coalescing operator?

Comment (tim-bezhashvyly) [inline src/SolrResponse.php]: Done. Good catch.

File: src/SolrDocumentBuilder.php (hunk 1 above)
Comment (tim-bezhashvyly) [inline src/SolrDocumentBuilder.php]: In all such cases the key is supposed to be unique.

Here for instance it is guaranteed by `SearchDocumentFieldCollection` class. See `\LizardsAndPumpkins\DataPool\SearchEngine\SearchDocument\SearchDocumentFieldCollection::fromArray`.

We can of course to apply additional guarding but I'm not sure if we should.

**Comment by tim-bezhashvyly** [comment]
All requests are fulfilled.

**Pull request merged**